gnovm/pkg/gnolang/machine.go

```diff
@@ -237,6 +268,69 @@ func (m *Machine) RunMemPackage(memPkg *std.MemPackage, save bool) (*PackageNode
 	return pn, pv
 }
 
+func filterDuplicates(fset *FileSet) {
+	defined := make(map[Name]struct{}, 128)
+	for _, f := range fset.Files {
+		// returns true if name already defined
```

Comment:
maybe flip, because true implies add name succeeded, implying the name didn't already exist.